gammapy/scripts/image_pipe.py

```diff
@@ -64,9 +66,34 @@ def make_counts(self, obs_id):
         obs = self.data_store.obs(obs_id=obs_id)
         events = obs.events.select_energy(self.energy_band)
         events = events.select_offset(self.offset_band)
-        counts_map.fill(events=events)
-        counts_map.data = counts_map.data.value
-        self.maps["counts"] = counts_map
+        if (len(events) != 0):
+            counts_map.fill(events=events)
+            counts_map.data = counts_map.data.value
+            self.maps["counts"] = counts_map
+        else:
+            self.maps["counts"] = counts_map
+
+    def _has_counts(self, obs_id):
```

Comment:
No. Imagine you have perfect background suppression. And a very week source where you only get one photon per 10 hours. Then to get the correct flux, you need to stack the exposure from the runs also where you had zero counts.

For HESS this can only occur if you pick a very small high-energy band.

But exposure has to be accumulated for *all* runs.

Makes sense?